### DISCUSSION TITLE
Latest kamailio IMS

### DISCUSSION THREAD

**Discussion by Sivanesh1992** [herlesupreeth/docker_open5gs#220] [General]
hi @herlesupreeth  as i seen we are using kamailio v 5.3 in our docker file .

in official gitub kamailio  project they are updated 5.7 version.

below link for reference 

https://github.com/kamailio/kamailio/branches

how to build latest version of kamailio in docker .

please guide me

**Comment by herlesupreeth** [comment]
Its not that straightforward unfortunately since I use the kamailio fork (https://github.com/herlesupreeth/kamailio) with many fixes rather than upstream kamailio repo (https://github.com/kamailio/kamailio). And, porting all the fixes to latest version is bit of pain as the upstream repo has changed a lot.

But, can you elaborate on why you would need version 5.7 of kamailio?? As its not adding any new functionality compared to v5.3

**Reply by Sivanesh1992** [reply]
as i discussed with one user in discord open5gs chat . he  tested vonr by using kamailio ims server. 

he only told latest version is supporting vonr .thats why i am planning to upgrade

**Reply by herlesupreeth** [reply]
To my knowledge no support for SBI interface was added in kamailio repo. So its not possible to test VoNR.

**Reply by herlesupreeth** [reply]
Can you point me to the chat in open5gs discord so that I can take a look?

**Reply by Sivanesh1992** [reply]
ya sure

**Reply by herlesupreeth** [reply]
@Sivanesh1992 I was able to test VoNR registration with iPhone 13 Pro by disabling Rx interface in P-CSCF and it worked. I am not sure whether the calling works (I doubt it since some phones are strict about having QoS for call). You can find more information here - https://github.com/herlesupreeth/docker_open5gs/discussions/248#discussioncomment-7392233

NOTE: If you are not using iPhone then you may not need SUPI concealment.

**Reply by Sivanesh1992** [reply]
Hi @herlesupreeth  happy to here this news. Let me test and update you. If i am using latest code also i need to disable RX interface in p-cscf  or not?

**Reply by herlesupreeth** [reply]
You need to just pull the latest changes and use the docker-compose file `sa-vonr-deploy.yaml` . Rx will be automatically disabled when using 5G SA

**Reply by Sivanesh1992** [reply]
Thankyou @herlesupreeth  let me check and update you
